client/app/scripts/superdesk/notification/notification.js

```diff
@@ -30,19 +34,67 @@
             return;
         }
 
-        var ws = new WebSocket(config.server.ws);
-
-        ws.onmessage = function(event) {
-            var msg = angular.fromJson(event.data);
-            $rootScope.$broadcast(msg.event, msg.extra);
-            if (_.contains(ReloadEvents, msg.event)) {
-                $rootScope.$broadcast('reload', msg);
-            }
+        var connect = function() {
+            ws = new WebSocket(config.server.ws);
+            bindEvents();
         };
 
-        ws.onerror = function(event) {
-            console.error(event);
+        var bindEvents = function() {
+            ws.onmessage = function(event) {
+                var msg = angular.fromJson(event.data);
+                $rootScope.$broadcast(msg.event, msg.extra);
+                if (_.contains(ReloadEvents, msg.event)) {
+                    $rootScope.$broadcast('reload', msg);
+                }
+            };
+
+            ws.onerror = function(event) {
+                console.error('in error ws.onerror');
+                console.error(event);
+            };
+
+            ws.onopen = function(event) {
+                $interval.cancel(connectTimer);
+                $rootScope.$broadcast('connected');
+            };
+
+            ws.onclose = function(event) {
+                $rootScope.$broadcast('disconnected');
+                $interval.cancel(connectTimer);
+                connectTimer = $interval(function() {
+                    if (ws) {
+                        connect();  // Retry to connect for every TIMEOUT interval.
+                    }
+                }, TIMEOUT);
```

Comment:
By default, `$interval` invokes the given function within an `$apply()`, triggering a new digest cycle. The latter is not needed in our case, thus we can prevent it from happening by passing the corresponding parameter to `$interval()`.